src/ksm/client/distributed_pod_label_discovery.go

```diff
@@ -70,11 +69,12 @@ func (p *distributedPodLabelDiscoverer) Discover(timeout time.Duration) ([]clien
 }
 
 // NewDistributedPodLabelDiscoverer creates a new KSM discoverer that will find KSM pods using k8s labels
-func NewDistributedPodLabelDiscoverer(ksmPodLabel string, nodeIP string, logger log.Logger, k8sClient client.Kubernetes) client.MultiDiscoverer {
+func NewDistributedPodLabelDiscoverer(ksmPodLabel, ksmNamespace, nodeIP string, logger log.Logger, k8sClient client.Kubernetes) client.MultiDiscoverer {
```

Comment:
So many args :(

Reply:
Yes... Want me to refactor that before the addition?

Reply:
I think it might be worth a try, if the impact it's not too big. It's up to you :)

Reply:
Done in #172